src/compile/scale/type.ts

```diff
@@ -157,7 +160,11 @@ export function fieldDefMatchScaleType(specifiedType: ScaleType, fieldDef: Field
     if (!fieldDef.timeUnit) {
       return contains([ScaleType.TIME, ScaleType.UTC, undefined], specifiedType);
     } else {
-      return contains([ScaleType.TIME, ScaleType.UTC, undefined], specifiedType) || hasDiscreteDomain(specifiedType);
+      if (isUTCTimeUnit(fieldDef.timeUnit)) {
+        return contains([ScaleType.UTC, undefined], specifiedType) || hasDiscreteDomain(specifiedType);
```

Comment:
Looks like we can refactor this method a bit (later in another PR) ==> See https://github.com/vega/vega-lite/issues/2347